src/main/java/com/cloudbees/jenkins/support/impl/NetworkInterfaces.java

```diff
@@ -0,0 +1,144 @@
+import hudson.security.Permission;
+import hudson.util.HexBinaryConverter;
+import jenkins.model.Jenkins;
+import org.apache.commons.codec.binary.Hex;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+/**
+ * @author schristou88
+ */
+@Extension
+public class NetworkInterfaces extends Component {
+    private final WeakHashMap<Node, String> networkInterfaceCache = new WeakHashMap<Node, String>();
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @Override
+    @NonNull
+    public String getDisplayName() {
+        return "Networking Interface";
+    }
+
+    @Override
+    public void addContents(@NonNull Container result) {
+        result.add(
+                new Content("nodes/master/networkInterface.md") {
+                    @Override
+                    public void writeTo(OutputStream os) throws IOException {
+                        os.write(getNetworkInterface(Jenkins.getInstance()).getBytes());
+                    }
+                }
+        );
+
+        for (final Node node : Jenkins.getInstance().getNodes()) {
+            result.add(
+                    new Content("nodes/slave/" + node.getNodeName() + "/networkInterface.md") {
+                        @Override
+                        public void writeTo(OutputStream os) throws IOException {
+                            os.write(getNetworkInterface(node).getBytes());
+                        }
+                    }
+            );
+        }
+    }
+
+    public String getNetworkInterface(Node node) throws IOException {
+        return AsyncResultCache.get(node,
+                networkInterfaceCache,
+                new GetNetworkInterfaces(),
+                "network interfaces",
+                "N/A: No connection to node, or no cache.");
+    }
+
+    private static final class GetNetworkInterfaces implements Callable<String, RuntimeException> {
+        public String call() {
+            StringWriter bos = new StringWriter();
+            PrintWriter pw = new PrintWriter(bos);
+
+            try {
+                Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+                while (networkInterfaces.hasMoreElements()) {
+                    pw.println("-----------");
+
+                    NetworkInterface ni = networkInterfaces.nextElement();
+                    pw.println(" * Name " + ni.getDisplayName());
+
+                    byte[] hardwareAddress = ni.getHardwareAddress();
+
+                    // Do not have permissions or address does not exist
+                    if (hardwareAddress == null || hardwareAddress.length == 0)
```

Comment:
As jenkins may only be bound to the loopback I think we should not break but just not dump the hardware address (or dump something special)